File: Backend/IELTS_PRACTICE/Models/Question.cs
namespace IELTS_PRACTICE.Models
{
    public class Question
    {
        public int Id { get; set; }
        public String QuestionType { get; set; }
        public String Content { get; set; }
        public String CorrectAnswer {  get; set; }
        public String Choices {  get; set; }
        public String Explanation { get; set; }
        public int ParentId { get; set; }
        public int TypeId { get; set; }
        public int TestId { get; set; }
        public String Link { get; set; }

        //Navigation
        public Type Type { get; set; }
        public Test Test { get; set; }
    }
}

File: Frontend/ielts-app/src/types/Question.ts
// same as Full Detail
export interface Question {
    id: number;
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    parentId: number;
    testId: number;
    order: number;
    link: string;
}

export interface QuestionGetReturn {
    id: number;
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    testId: number;
    order: number;
}

export interface QuestionParagraphReturn {
    id: number;
    content: string;
    testId: number;
}

export interface QuestionToCreate {
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    parentId: number;
    testId: number;
    link: string;
    order: number;
}

// Should be the same as Question
export interface QuestionPostReturn {
    id: number;
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    parentId: number;
    testId: number;
    order: number;
    link: string;
}

export interface QuestionToUpdate {
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    link: string;
    parentId: number;
    order: number;
}

export interface QuestionPutReturn {
    id: number;
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    testId: number;
    order: number;
}

// Same as Question, has all fields
export interface QuestionFullDetail {
    id: number;
    questionType: string;
    content: string;
    correctAnswer: string;
    choices: string;
    explanation: string;
    testId: number;
    order: number;
    parentId: number;
    link: string;
}